### PR TITLE
[AIRFLOW-5588] Add Celery's architecture diagram

### PR DESCRIPTION
**Pull request by mik-laj**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by mik-laj** [comment]
Preview: http://onerous-meeting.surge.sh/executor/celery.html

File: docs/executor/celery.rst [hunk 1]
```diff
@@ -72,3 +72,74 @@ Some caveats:
...
+            scheduler->queue_result_backend[label="10"]
+            scheduler->queue_broker[label="11"]
+        }
+    }
+
+Airflow consist of several components:
+
+* **Workers** - Execute the assigned tasks
+* **Scheduler** - Responsible for adding the necessary tasks to the queue
+* **Web server** - Server HTTP provides access to DAG/task status information
```
Comment (milton0825) [inline docs/executor/celery.rst]:
```suggestion
* **Web server** - HTTP Server provides access to DAG/task status information
```

File: docs/executor/celery.rst [hunk 2]
```diff
@@ -72,3 +72,74 @@ Some caveats:
...
+* **Queue** - Queue mechanism provided by Celery
+
+Please note that the queue at Celery consists of two components:
+
+* **Broker** - Stores commands for execution
+* **Result backend** - Stores status of completed command
+
+The components communicate with each other in many places
+
+* [1] **Scheduler** --> **Workers** - Fetchs task execution logs
```
Comment (milton0825) [inline docs/executor/celery.rst]: I thought it is web server that fetches execution logs from worker and display it on the UI. Correct me if I am wrong

**Comment by milton0825** [comment]
Nice diagram

**Comment by mik-laj** [comment]
@milton0825 Thanks for review.
Updated preview: http://hesitant-pig.surge.sh/executor/celery.html

File: docs/executor/celery.rst (hunk 2 above)
Comment (mik-laj) [reply docs/executor/celery.rst]: I checked it out. This is true. Webserver fetches logs.
https://github.com/apache/airflow/blob/d719e1f/airflow/www/views.py#L554-L572
https://github.com/apache/airflow/blob/d719e1fd6705a93a0dfefef4b46478ade5e006ea/airflow/utils/log/file_task_handler.py#L110-L132 
I updated tihis PR.

File: docs/executor/celery.rst (hunk 1 above)
Comment (mik-laj) [reply docs/executor/celery.rst]: Updated

**Review by milton0825 — APPROVED** [review]

**Review by feng-tao — APPROVED** [review]

**Review by kaxil — COMMENTED** [review]
Good work @mik-laj

**Pull request merged**